lib/active_interaction/filter.rb

```diff
@@ -90,15 +90,16 @@ def initialize(name, options = {}, &block)
     #   # => ActiveInteraction::InvalidDefaultError: example: 0
     #
     # @param value [Object]
+    # @param interaction [Base, nil]
     #
     # @return [Object]
     #
     # @raise (see #cast)
     # @raise (see #default)
-    def clean(value)
-      value = cast(value)
+    def clean(value, interaction)
```

Comment:
I thought about `binding`, but it's not actually a [`Binding`](http://ruby-doc.org/core-2.2.3/Binding.html) object. I prefer `interaction` over `context` because it's *always* an interaction. "Context" may kind of explain what it's doing, but "interaction" explains what it *is*. 

That being said, it might be better to pass a duplicate of the interaction's binding instead of the interaction itself. That would prevent you from modifying the interaction inside the default proc.